pksmanager/busmanager/busmanager-infrastructure/src/main/java/pl/kompikownia/pksmanager/busmanager/infrastructure/repository/jpa/InsurancesRepositoryImpl.java

```diff
@@ -0,0 +1,86 @@
+package pl.kompikownia.pksmanager.busmanager.infrastructure.repository.jpa;
+
+import com.querydsl.jpa.impl.JPADeleteClause;
+import com.querydsl.jpa.impl.JPAQuery;
+import lombok.RequiredArgsConstructor;
+import lombok.val;
+import org.springframework.stereotype.Repository;
+import pl.kompikownia.pksmanager.busmanager.business.projection.InsurancesProjection;
+import pl.kompikownia.pksmanager.busmanager.business.repository.InsurancesRepository;
+import pl.kompikownia.pksmanager.busmanager.infrastructure.entity.BusEntity;
+import pl.kompikownia.pksmanager.busmanager.infrastructure.entity.InsurancesEntity;
+import pl.kompikownia.pksmanager.busmanager.infrastructure.repository.port.InsurancesCrudRepository;
+
+import javax.persistence.EntityManager;
+import javax.persistence.PersistenceContext;
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static pl.kompikownia.pksmanager.busmanager.infrastructure.entity.QInsurancesEntity.insurancesEntity;
+
+@Repository
+@RequiredArgsConstructor
+public class InsurancesRepositoryImpl implements InsurancesRepository {
+
+    @PersistenceContext
+    private EntityManager em;
+
+    private final InsurancesCrudRepository insurancesCrudRepository;
+
+
+    @Override
+    public InsurancesProjection save(InsurancesProjection insurancesProjection) {
+        val entityToPersist = InsurancesEntity.of(em,insurancesProjection);
+        val parentEntity = em.getReference(BusEntity.class, insurancesProjection.getBusId());
+        em.merge(entityToPersist);
+        em.merge(parentEntity);
+        em.flush();
+        return entityToPersist.toProjection();
+    }
+
+    @Override
+    public InsurancesProjection saveById(InsurancesProjection insurancesProjection, Long id) {
+        val entityToPersist = InsurancesEntity.of(em,insurancesProjection);
+        val parentEntity = em.getReference(BusEntity.class, id);
+        em.merge(entityToPersist);
+        em.merge(parentEntity);
```

Comment:
Analogicznie jak komentarz wyżej

Reply:
@Override
    @Transactional
    public BusStopProjection save(BusStopProjection busStopProjection) {
        val entityToPersist = BusStopEntity.of(em, busStopProjection);
        val parentEntity = em.getReference(ScheduleEntity.class, busStopProjection.getScheduleId());
        parentEntity.getBusStopEntities().add(entityToPersist);
        em.merge(entityToPersist);
        em.merge(parentEntity);
        return entityToPersist.toProjection();
    }